File: PCD_CF849_Div4/E_Negatives_and_Positives.py
t=int(input())
while(t>0):
  n=int(input())
  alist=input().split()
  absum=0
  min1=abs(int(alist[0]))
  count=0
  for i in alist:
    i=int(i)
    if(i<0):
      count=count+1
    absum+=abs(i)
    min1=min(min1,abs(i))
  if(count%2==0):
    print(absum)
  else:
    print(absum-(2*min1))
   
  t=t-1

File: PCD/PCD_CF849_Div4/E_Negatives_and_Positives.cpp
#include<bits/stdc++.h>
using namespace std;
#define int long long

void solve(){
    int n;
    cin>>n;
    
    int absum=0;
    int count=0;
    int min1=INT_MAX;
    
    for(int i=0;i<n;i++){
        int x;
        cin>>x;
        if(x<0){
            count++;
        }
        absum+=abs(x);
        min1=min(min1,abs(x));
    }

    if(count&1){
        cout<<absum-(2*min1)<<endl;
    }
    else{
        cout<<absum<<endl;
    }

}
signed main()
{
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}
